src/Datadog.Trace/Configuration/GlobalSettings.cs

```diff
@@ -111,29 +112,71 @@ internal static CompositeConfigurationSource CreateDefaultConfigurationSource()
 #endif
             };
 
-            string currentDirectory = System.Environment.CurrentDirectory;
+            if (TryLoadJsonConfigurationFile(configurationSource, out var jsonConfigurationSource))
+            {
+                configurationSource.Add(jsonConfigurationSource);
+            }
 
-#if NETFRAMEWORK
-            // on .NET Framework only, use application's root folder
-            // as default path when looking for datadog.json
-            if (System.Web.Hosting.HostingEnvironment.IsHosted)
+            return configurationSource;
+        }
+
+        private static bool TryLoadJsonConfigurationFile(IConfigurationSource configurationSource, out IConfigurationSource jsonConfigurationSource)
+        {
+            try
             {
-                currentDirectory = System.Web.Hosting.HostingEnvironment.MapPath("~");
+                string currentDirectory = System.Environment.CurrentDirectory;
+
+                try
+                {
+                    if (TryLoadHostingEnvironmentPath(out var hostingPath))
+                    {
+                        currentDirectory = hostingPath;
+                    }
+                }
+                catch (Exception)
+                {
+                    // Unable to call into System.Web.dll
+                    // The configuration manager should not depend on a logger being bootstrapped yet
+                    // so do not do anything
+                }
+
+                // if environment variable is not set, look for default file name in the current directory
+                var configurationFileName = configurationSource.GetString(ConfigurationKeys.ConfigurationFileName) ??
+                                            configurationSource.GetString("DD_DOTNET_TRACER_CONFIG_FILE") ??
+                                            Path.Combine(currentDirectory, "datadog.json");
```

Comment:
This is just a tiny optimization, but if users set the config file path, we don't need to compute `currentDirectory` at all. Should we move that logic into a separate method that is only called if needed?
```suggestion
                                            Path.Combine(GetCurrentDirectory(), "datadog.json");
```

Reply:
In my refactoring in f4d4b7ba8db18e83463a602fee0b7564fcf45908 I also addressed this